server/neptune/gateway/server.go

```diff
@@ -263,6 +269,36 @@ func NewServer(config Config) (*Server, error) {
 		return nil, errors.Wrap(err, "initializing temporal client")
 	}
 
+	ghToken, err := githubCredentials.GetToken()
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching github token")
+	}
+	repoFetcher := &github.RepoFetcher{
+		DataDir:        config.DataDir,
+		Token:          ghToken,
+		GithubHostname: config.GithubHostname,
+		Logger:         ctxLogger,
+	}
+	hooksRunner := &preworkflow.HooksRunner{
+		GlobalCfg:    globalCfg,
+		HookExecutor: &preworkflow.HookExecutor{},
+	}
+	clientCreator, err := githubapp.NewDefaultCachingClientCreator(
+		config.AppCfg,
+		githubapp.WithClientMiddleware(
```

Comment:
can we move [middleware](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/neptune/workflows/internal/activities/github/middleware.go) out of the internal pkg? I duplicated it for now, but wanted to confirm it's okay for the workflows to use.

Reply:
yeah that seems reasonable to me